src/plugins/regmon/regmon.cpp

```diff
@@ -264,6 +281,135 @@ static event_response_t log_reg_value_hook_cb( drakvuf_t drakvuf, drakvuf_trap_i
     return log_reg_hook( drakvuf, info, key_handle_addr, value_name_addr, true );
 }
 
+static event_response_t log_reg_set_value_hook_cb( drakvuf_t drakvuf, drakvuf_trap_info_t* info )
+{
+
+    enum RegistryValueTypes
+    {
+        REG_NONE = 0
+        ,REG_SZ
+        ,REG_EXPAND_SZ
+        ,REG_BINARY
+        ,REG_DWORD
+        ,REG_DWORD_LITTLE_ENDIAN = REG_DWORD
+        ,REG_DWORD_BIG_ENDIAN
+        ,REG_LINK
+        ,REG_MULTI_SZ
+        ,REG_RESOURCE_LIST
+        ,REG_FULL_RESOURCE_DESCRIPTOR
+        ,REG_RESOURCE_REQUIREMENTS_LIST
+        ,REG_QWORD
+        ,REG_QWORD_LITTLE_ENDIAN = REG_QWORD
+    };
+
+    addr_t key_handle_addr = drakvuf_get_function_argument(drakvuf, info, 1);
+
+    if (!key_handle_addr)
+        return 0;
+
+    addr_t value_name_addr = drakvuf_get_function_argument(drakvuf, info, 2);
+    uint32_t type = drakvuf_get_function_argument(drakvuf, info, 4);
+    addr_t data_addr = drakvuf_get_function_argument(drakvuf, info, 5);
+    uint32_t data_size = drakvuf_get_function_argument(drakvuf, info, 6);
+
+    unicode_string_t* data_us = nullptr;
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+    ctx.addr = data_addr;
+
+    if ((type == REG_SZ) || (type == REG_LINK) || (type == REG_EXPAND_SZ))
+    {
+        vmi_lock_guard vmi_lg(drakvuf);
+        data_us = drakvuf_read_wchar_string(vmi_lg.vmi, &ctx);
+    }
+    else
+    {
+        std::vector<uint8_t> data_bytes(data_size, 0);
+        vmi_lock_guard vmi_lg(drakvuf);
```

Comment:
Is this is lock in the right scope here? I only see one `vmi_read` that's inside the following block.

Reply:
It is actually used one more time on line 365

Reply:
Hm, I would say it would still be better scoped to have the lock within the following block and then another lock in block around 365

Reply:
Done